shell/common/shell.cc

```diff
@@ -67,58 +67,57 @@ std::unique_ptr<Shell> Shell::CreateShellOnPlatformThread(
   // first because it has state that the other subsystems depend on. It must
   // first be booted and the necessary references obtained to initialize the
   // other subsystems.
-  fml::AutoResetWaitableEvent io_latch;
-  std::unique_ptr<ShellIOManager> io_manager;
+  std::promise<std::unique_ptr<ShellIOManager>> io_manager_promise;
+  auto io_manager = io_manager_promise.get_future();
```

Comment:
Here and elsewhere, please append `_future` to the ivars or locals that are now futures.

Reply:
done